compiler/bindings/python/iree/compiler/tools/import_onnx/__main__.py

```diff
@@ -132,6 +139,29 @@ def parse_arguments(argv=None) -> argparse.Namespace:
         " Defaults to the directory of the input file.",
         type=Path,
     )
+    parser.add_argument(
+        "--opset-version",
+        help="Allows specification of a newer opset_version to update the model"
+        " to before importing to MLIR. This can sometime assist with shape inference.",
+        type=int,
+    )
```

Comment:
Should we default this to a value like 17? We set various versions in test suites:
* https://github.com/iree-org/iree-test-suites/blob/26ee0c25a1ed1ae83aaace91a7f380f8243b4056/onnx_ops/import_onnx_tests.py#L23-L24
* https://github.com/nod-ai/SHARK-TestSuite/blob/9aa03f5f0aceac5669fff3f663b55a27490a2c15/e2eshark/onnx/operators/AveragePool/model.py#L54-L56
* https://github.com/nod-ai/SHARK-TestSuite/blob/9aa03f5f0aceac5669fff3f663b55a27490a2c15/alt_e2eshark/onnx_tests/helper_classes.py#L23-L24

(or could split the version upgrade code into its own PR, separate from the external parameters handling)

Reply:
Bump on this comment. Want to set a default?

> This can sometime assist with shape inference

It does more than that - many of the conversion patterns in torch-mlir set a minimum opset version far above what is possible to find in archived .onnx files (like those in https://github.com/onnx/onnx/tree/main/onnx/backend/test/data and https://github.com/onnx/models)

Take https://onnx.ai/onnx/operators/onnx__BatchNormalization.html for example - that has been defined since ONNX version `1`, but the pattern for converting it from onnx to torch specifies a minimum version of `15`: https://github.com/llvm/torch-mlir/blob/4c1518d365823c3f01d388f4d0f84112b9aa7808/lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp#L343

That's why I think we should pick a minimum version and always upgrade, or try to convert with the given version and automatically try upgrading if that fails first.

(but as stated in my first comment, the version upgrade code can be separated from the external parameter code - try to keep PRs focused on a single feature)